### PR TITLE
Huvu/t5 nemo2.0 peft

### PR DESCRIPTION
**Pull request by huvunvidia**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by cuichenx — COMMENTED** [review]
LGTM on for code changes! 
For CI @pablo-garay can comment further but I think we might want to use /tmp dir going forward?

File: .github/workflows/cicd-main.yml [hunk 1]
```diff
@@ -5182,6 +5182,22 @@ jobs:
...
+    uses: ./.github/workflows/_test_template.yml
+    if: contains(fromJSON(needs.cicd-test-container-setup.outputs.test_to_run), 'L2_NeMo_2_T5_LoRA') || needs.cicd-test-container-setup.outputs.all == 'true'
+    with:
+      RUNNER: self-hosted-azure
+      SCRIPT: |
+        NVTE_FUSED_ATTN=0 NVTE_FLASH_ATTN=0 python tests/collections/llm/megatron_t5_finetuning.py \
+        --devices=2 \
+        --max-steps=250 \
+        --peft=lora \
+        --experiment-dir=tests/collections/llm/t5_peft_results/${{ github.run_id }} \
```
Comment (pablo-garay) [inline .github/workflows/cicd-main.yml:5196]: Can we use the /tmp folder for output files here ? This is a new practice we started recently 
E,g,
tmp/t5_peft_results

File: .github/workflows/cicd-main.yml [hunk 2]
```diff
@@ -5182,6 +5182,22 @@ jobs:
...
+        --checkpoint-path=/home/TestData/nlp/megatron_t5/220m/nemo2.0_t5_220m_150steps
+      AFTER_SCRIPT: |
+        rm -rf tests/collections/llm/t5_peft_results/${{ github.run_id }}
```
Comment (pablo-garay) [inline .github/workflows/cicd-main.yml:5199]: If using /tmp we dont need this anymore

Comment (huvunvidia) [reply .github/workflows/cicd-main.yml:5199]: Will add it in a separate PR.

File: .github/workflows/cicd-main.yml (hunk 1 above)
Comment (huvunvidia) [reply .github/workflows/cicd-main.yml:5196]: Will add it in a separate PR.

**Review by cuichenx — APPROVED** [review]

**Review by pablo-garay — APPROVED** [review]

**Pull request merged**